spec/unit/logstash_output_spec.rb

```diff
@@ -30,8 +30,49 @@
   describe "plugin register" do
     let(:registered_plugin) { plugin.tap(&:register) }
 
+    describe "normalize hosts" do
+
+      it "applies default https scheme and 9800 port" do
+        expect(registered_plugin.hosts.eql?(::LogStash::Util::SafeURI.new("https://127.0.0.1:9800/"))).to be_truthy
+      end
+
+      describe "SSL disabled" do
+        let(:config) { super().merge("ssl_enabled" => false) }
+
+        it "causes HTTP scheme" do
+          expect(registered_plugin.hosts.eql?(::LogStash::Util::SafeURI.new("http://127.0.0.1:9800/"))).to be_truthy
+        end
+      end
+
+      describe "custom port" do
+        let(:config) { super().merge("hosts" => "127.0.0.1:9808") }
+
+        it "will be applied" do
+          expect(registered_plugin.hosts.eql?(::LogStash::Util::SafeURI.new("https://127.0.0.1:9808/"))).to be_truthy
+        end
+      end
+
+      describe "`ssl_enabled => false` but HTTPS scheme is used" do
+        let(:config) { super().merge("ssl_enabled" => false, "hosts" => %w(https://127.0.0.1:9808)) }
```

Comment:
http(s) is an implementation detail that should NOT be exposed to the `hosts` config.

We should accept _only_ colon-separated hostname-port pairs, formally:

~~~ antlr
hostsEntry: host ':' port
          ;

host: ipAddress
    | hostname
    ;

ipAddress:  ipv4Address
          | '[' ipv6Address ']'
          ;

# ...
# host is a valid domain name (e.g. `myserver1.example.com`)
# ipv4Address is dot-decimal representation (e.g. `127.0.0.1`)
# ipv6Address is any RFC4291§2.2 text representation of an ipv6 address (e.g., `::1`, `2001:0db8:85a3:0000:0000:8a2e:0370:7334`, `::ffff:93.184.216.34`)
~~~

Reply:
Ah, yes the protocol is an implementation detail... I will make a changed based on the `adapter` suggestion you mentioned. Thank you for catching this up!